src/main.rs

```diff
@@ -2,164 +2,44 @@
-
-type VecStore<E> = merkletree::store::VecStore<E>;
-pub type MerkleTree<T, A> = merkletree::merkle::MerkleTree<T, A, VecStore<T>>;
-
-// use a file as an io::Reader but pad out extra length at the end with zeros up
-// to the padded size
-struct PadReader {
-    file: File,
-    fsize: usize,
-    padsize: usize,
-    pos: usize,
-}
-
-impl io::Read for PadReader {
-    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        /*
-        if self.pos == self.fsize {
-          print!("reached file size, now padding ...")
-        }
-        */
-        if self.pos >= self.fsize {
-            for i in 0..buf.len() {
-                buf[i] = 0;
-            }
-            let cs = cmp::min(self.padsize - self.pos, buf.len());
-            self.pos = self.pos + cs;
-            /*
-            if cs < buf.len() {
-              print!("done with file ...")
-            }
-            */
-            Ok(cs)
-        } else {
-            let cs = self.file.read(buf)?;
-            self.pos = self.pos + cs;
-            Ok(cs)
-        }
-    }
-}
-
-// logic partly copied from Lotus' PadReader which is also in go-fil-markets
-// figure out how big this piece will be when padded
-fn padded_size(size: u64) -> u64 {
-    let logv = 64 - size.leading_zeros();
-    let sect_size = (1 as u64) << logv;
-    let bound = u64::from(UnpaddedBytesAmount::from(SectorSize(sect_size)));
-    if size <= bound {
-        return bound;
-    }
-    return u64::from(UnpaddedBytesAmount::from(SectorSize(1 << (logv + 1))));
-}
-
-// from rust-fil-proofs/src/storage_proofs/pieces.rs
-fn local_generate_piece_commitment_bytes_from_source<H: Hasher>(
-    source: &mut dyn Read,
-    padded_piece_size: usize,
-) -> anyhow::Result<Fr32Ary> {
-    ensure!(padded_piece_size > 32, "piece is too small");
-    ensure!(padded_piece_size % 32 == 0, "piece is not valid size");
-
-    let mut buf = [0; NODE_SIZE];
-    use std::io::BufReader;
-
-    let mut reader = BufReader::new(source);
-
-    let parts = (padded_piece_size as f64 / NODE_SIZE as f64).ceil() as usize;
-
-    let tree = MerkleTree::<H::Domain, H::Function>::try_from_iter((0..parts).map(|_| {
-        reader.read_exact(&mut buf)?;
-        <H::Domain as Domain>::try_from_bytes(&buf)
-    }))?;
-
-    let mut comm_p_bytes = [0; NODE_SIZE];
-    let comm_p = tree.root();
-    comm_p.write_bytes(&mut comm_p_bytes)?;
-
-    Ok(comm_p_bytes)
-}
+extern crate simple_logger;
 
 fn usage() {
     print!("Usage: commp [-fp|-sp|-spl] <file>\n");
 }
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    simple_logger::init_with_level(log::Level::Info)?;
+
     let args: Vec<String> = env::args().collect();
     if args.len() < 3 {
         usage();
         return Err(From::from("Not enough arguments".to_string()));
     }
     let filename = &args[2];
-    let file = File::open(filename).expect("Unable to open file");
+    let mut file = File::open(filename).expect("Unable to open file");
```

Comment:
As your main function already returns a result, you can return an error here instead of panicking:

```rust
let mut file = std::fs::File::open(filename).map_err(|_| format!("Unable to open file: {}", filename))?;
```
or if returning the original error (`Os { code: 2, kind: NotFound, message: "No such file or directory" }`) is good enough simply

```rust
let mut file = File::open(filename)?;
```